databaseutil/databaseutil.go

```diff
@@ -94,6 +94,59 @@ func GetPasswordForUserWithEmailAddress(emailAddress string) ([]byte, error) {
 	return password, nil
 }
 
+func StoreNewNote(authorId int64, content string, creationTime time.Time) (int64, error) {
```

Comment:
I'm realizing more and more that I think I made a mistake when I said we should only use builtin types in this file. I shouldn't be looking at that authorId wondering if someone passed in an authorId or not

Reply:
To clarify, you mean you shouldn't wonder if the `authorId` argument is of type `int64` or of type `UserId`? 

Honestly this file only gets waaay worse the more calls to the database we make. If you look at my later branches this file gets really out of hand. Personally i think we should consider adding an ORM package. Something that translates objects to sql and back. I think it would make our code look alot cleaner. But i also think that's a discussion we should make after we're done with the other CRs and we have all the pieces in front of us.

Reply:
The only ORM i've ever used in that way is Ruby on Rails, so maybe i can't judge that pattern fairly. But I think the issue with this file isn't the fact that it has a lot of sql code. The issue is I think my idea was that this would be some sort of idealistic pure replacement of creating a builtin postgres function, which i was trying to avoid. I think that dream is dead, because it's only making me confused when i look at these function definitions and wonder why they've lost their type specificity